### DISCUSSION TITLE
Error when registering a Singleton

### DISCUSSION THREAD

**Discussion by AngeloLazzari** [PrismLibrary/Prism#3255] [.NET MAUI]
We are migrating our old xamarin app to the new MAUI one.

In the xamarin verision we used this code to instantiate the singleton 
`builder.Services.AddSingleton<IMessage>();
builder.Services.AddSingleton<ILodingPageService>();`

but compiling this error comes out:
![error2](https://github.com/user-attachments/assets/120214a7-7726-4215-abdc-dd8f7bc8660d)

So we tried to use the RegisterSingleton of your great prism library as shown in the image but we have the same but from the builder...

![errro](https://github.com/user-attachments/assets/efc8c4e7-ccd0-418b-9665-1d6ca903f423)

How do we have to treat that code now that the simple MAUI singleton is not working like in xamarin?

Thank you very much
Angelo

**Comment by dansiegel** [comment]
Yes if you read the error it tells you exactly what the problem is you are registering the interface with no concrete implementation therefore it will have an issue with the registration because there is no way that it could create a new instance.

```cs
// Won't work...
builder.Services.AddSingleton<IMessage>(); 
builder.Services.AddSingleton<ILodingPageService>();

// Would work...
builder.Services.AddSingleton<IMessage, MessageService>();
builder.Services.AddSingleton<ILodingPageService, LoadingPageService>();
```

**Reply by AngeloLazzari** [reply]
Thank you very much for the repluy, but, why in xamarin it was working? thank you

**Reply by dansiegel** [reply]
That never would have worked the only difference is that your old Xamarin app was using an older version of DryIoc which may not have thrown an error about it but that code would never really work.

**Reply by AngeloLazzari** [reply]
ok ok thanks, that is why. Thank you very much Dan.
